### PR TITLE
feat(rxjs): create a set of rxjs operators

### PR DESCRIPTION
**Pull request by tomalaforge**
closing issue #41

### CODE REVIEW

File: docs/src/content/docs/utilities/filter-nil.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+title: filter nil RxJs operator
+description: ngxtension/filter-array
```
Comment (nartc) [inline docs/src/content/docs/utilities/filter-nil.md:3]: nit: wrong description>?

File: libs/ngxtension/filter-array/src/filter-array.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { map, pipe } from 'rxjs';
+
+export const filterArray = <T>(filterFn: (item: T) => boolean) => {
+	return pipe(map((array: T[]) => array.filter((item) => filterFn(item))));
```
Comment (nartc) [inline libs/ngxtension/filter-array/src/filter-array.ts:4]: suggestion: I'm pretty sure `pipe()` isn't needed.

File: libs/ngxtension/filter-nil/src/map-array.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { filter, pipe } from 'rxjs';
+
+export const filterNil = <T>() =>
+	pipe(
```
Comment (nartc) [inline libs/ngxtension/filter-nil/src/map-array.ts:4]: suggestion: `pipe()` isn't needed

File: libs/ngxtension/map-array/src/map-array.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { map, pipe } from 'rxjs';
+
+export const mapArray = <T, R>(mapFn: (item: T) => R) => {
+	return pipe(map((array: T[]) => array.map((item) => mapFn(item))));
```
Comment (nartc) [inline libs/ngxtension/map-array/src/map-array.ts:4]: suggestion: `pipe()` isn't needed

File: libs/ngxtension/project.json [hunk 5]
```diff
@@ -39,7 +39,11 @@
 				"lintFilePatterns": [
 					"libs/ngxtension/**/*.ts",
 					"libs/ngxtension/**/*.html",
-					"libs/ngxtension/package.json"
+					"libs/ngxtension/package.json",
+					"libs/ngxtension/filter-array/**/*.ts",
+					"libs/ngxtension/filter-array/**/*.html",
+					"libs/ngxtension/filter-nil/**/*.ts",
+					"libs/ngxtension/filter-nil/**/*.html"
```
Comment (nartc) [inline libs/ngxtension/project.json:46]: comment: remove these

**Review by nartc — CHANGES_REQUESTED** [review]
Thank you for your contribution. A couple of nits

**Comment by nartc** [comment]
@all-contributors please add @tomalaforge for code

**Comment by tomalaforge** [comment]
thanks @nartc for reviewing my PR. 

Changes made

**Review by nartc — APPROVED** [review]

**Pull request merged**